app/controllers/paypal_accounts_controller.rb

```diff
@@ -214,7 +214,18 @@ def flash_error_and_redirect_to_settings(error_response: nil, error_msg: nil)
       end
 
     flash[:error] = error_msg
-    redirect_to new_paypal_account_settings_payment_path(@current_user.username)
+    redirect_to error_redirect_to
+  end
+
+  def error_redirect_to
```

Comment:
The rationale is this:

`redirect_to redirect_destination`

And then the implementation can be:

```ruby
def redirect_destination
  {action: :whatever}
end

# or

def redirect_destination
  new_settings_whatever_path(@current_user.id)
end
```